kotlinpoet/src/main/java/com/squareup/kotlinpoet/FunSpec.kt

```diff
@@ -245,6 +245,25 @@ public class FunSpec private constructor(
     return null
   }
 
+  private fun CodeBlock.returnsWithoutLinebreak(): CodeBlock {
+    val originCodeBlockBuilder = toBuilder()
+    val returnWithSpace = CodeBlock.RETURN_WITH_SPACE
+    originCodeBlockBuilder.formatParts.clear()
+    formatParts.mapTo(originCodeBlockBuilder.formatParts) { formatPart ->
```

Comment:
TBH I have a hard time understanding what this block does, can it be simplified? Essentially what we want to do is (in pseudocode):

```
if (body.startsWith(RETURN_EXPRESSION_BODY_PREFIX_SPACE)) {
  body = body.replace(RETURN_EXPRESSION_BODY_PREFIX_SPACE, RETURN_EXPRESSION_BODY_PREFIX_NBSP)
}
```

Helper methods inside `CodeBlock`, such as `withoutPrefix()`, `trim()`, `replaceAll()` can help implement that.

Reply:
We don't need to support trimming. If the person is generating code which has leading spaces or multiple spaces between the actual return expression then they can fix that.